common/metrics/interfaces.go

```diff
@@ -18,6 +18,8 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+//go:generate mockgen -package $GOPACKAGE -source $GOFILE -destination client_mock.go
```

Comment:
I'd say it should be easier to use tally test scope to ensure metrics emission instead of using a mock in this case.

Reply:
Changed to tally testscope